rector.php

```diff
@@ -0,0 +1,120 @@
+    $rectorConfig->bootstrapFiles([
+        realpath(getcwd()) . '/vendor/codeigniter4/framework/system/Test/bootstrap.php',
+    ]);
+
+    if (is_file(__DIR__ . '/phpstan.neon.dist')) {
+        $rectorConfig->phpstanConfig(__DIR__ . '/phpstan.neon.dist');
+    }
+
+    // Set the target version for refactoring
+    $rectorConfig->phpVersion(PhpVersion::PHP_74);
+
+    // Auto-import fully qualified class names
+    $rectorConfig->importNames();
+
+    // Are there files or rules you need to skip?
+    $rectorConfig->skip([
+        __DIR__ . '/app/Views',
+
+        JsonThrowOnErrorRector::class,
+        StringifyStrNeedlesRector::class,
+
+        // Note: requires php 8
+        RemoveUnusedPromotedPropertyRector::class,
+
+        // Ignore tests that might make calls without a result
+        RemoveEmptyMethodCallRector::class => [
+            __DIR__ . '/tests',
+        ],
+
+        // Ignore files that should not be namespaced
+        NormalizeNamespaceByPSR4ComposerAutoloadRector::class => [
+            __DIR__ . '/app/Helpers',
+        ],
+
+        // May load view files directly when detecting classes
+        StringClassNameToClassConstantRector::class,
+
+        // May be uninitialized on purpose
+        AddDefaultValueForUndefinedVariableRector::class,
+    ]);
+    $rectorConfig->rule(SimplifyUselessVariableRector::class);
+    $rectorConfig->rule(RemoveAlwaysElseRector::class);
+    $rectorConfig->rule(CountArrayToEmptyArrayComparisonRector::class);
+    $rectorConfig->rule(ForToForeachRector::class);
+    $rectorConfig->rule(ChangeNestedForeachIfsToEarlyContinueRector::class);
+    $rectorConfig->rule(ChangeIfElseValueAssignToEarlyReturnRector::class);
+    $rectorConfig->rule(SimplifyStrposLowerRector::class);
+    $rectorConfig->rule(CombineIfRector::class);
+    $rectorConfig->rule(SimplifyIfReturnBoolRector::class);
+    $rectorConfig->rule(InlineIfToExplicitIfRector::class);
+    $rectorConfig->rule(PreparedValueToEarlyReturnRector::class);
+    $rectorConfig->rule(ShortenElseIfRector::class);
+    $rectorConfig->rule(SimplifyIfElseToTernaryRector::class);
+    $rectorConfig->rule(UnusedForeachValueToArrayKeysRector::class);
+    $rectorConfig->rule(ChangeArrayPushToArrayAssignRector::class);
+    $rectorConfig->rule(UnnecessaryTernaryExpressionRector::class);
+    $rectorConfig->rule(AddPregQuoteDelimiterRector::class);
+    $rectorConfig->rule(SimplifyRegexPatternRector::class);
+    $rectorConfig->rule(FuncGetArgsToVariadicParamRector::class);
+    $rectorConfig->rule(MakeInheritedMethodVisibilitySameAsParentRector::class);
+    $rectorConfig->rule(SimplifyEmptyArrayCheckRector::class);
+    $rectorConfig->rule(NormalizeNamespaceByPSR4ComposerAutoloadRector::class);
+    $rectorConfig
+        ->ruleWithConfiguration(TypedPropertyRector::class, [
+            // Set to false if you use in libraries, or it does create breaking changes.
+            TypedPropertyRector::INLINE_PUBLIC => false,
```

Comment:
This is not a library. We don't need to restrict.

```suggestion
            TypedPropertyRector::INLINE_PUBLIC => true,
```

Reply:
This was adding types to all the Config files in **app/Config/** which means they deviate from the framework versions. While this is acceptable by LSP it makes patching a lot harder because every single file shows as "changed". I decided the benefit of having typed properties wasn't worth the extra upkeep effort. (This is how I am handling many projects, not just this one.)

Reply:
I suppose a middle ground would be to apply all the changes this would make then restore the Config files and `false` value, so at least we were getting types elsewhere.

Reply:
I forgot that there may be no type in the framework's Config file.
I got it.